### PR TITLE
Zauker refactor (attempt 2)

### PR DESCRIPTION
**Pull request by TheBoondock**
## About The Pull Request
Zauker currently perform its effects on breathe in lungs.dm which I've been informed is quite unreliable compare to chem
Creating a seperate branch since my branch for #65808 went haywire and i couldnt push changes

## Why It's Good For The Game
Refactor zauker to deal its effects on reagent instead to be more reliable

## Changelog

:cl:
refactor: Zauker now uses reagent to perform its effects on breath
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -1524,6 +1524,22 @@
...
+	color = "90560B"
+	taste_description = "bitter"
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED|REAGENT_NO_RANDOM_RECIPE
+
+/datum/reagent/zauker/on_mob_life(mob/living/L, delta_time, times_fired)
+	. = ..()
+	L.adjustBruteLoss(25 * REM * delta_time, FALSE)
+	L.adjustOxyLoss(5 * REM * delta_time, FALSE)
+	L.adjustFireLoss(8 * REM * delta_time, FALSE)
+	L.adjustToxLoss(8 * REM * delta_time, FALSE)
```
Comment (MrMelbert) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1541]: Don't let this be synthesizable or else botany will have a field day

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Breaths are done every 8 seconds / 4 life ticks, you must make sure this is either metabolized at the same rate or the numbers are adjusted accordingly to a faster metabolism

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 1 above)
Comment (TheBoondock) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:1541]: I think if botany manages to roll this, they deserve their lottery

Comment (timothymtorres) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm:1541]: Would this mean people could remove the reagent from someone's bloodstream and use it as poison?

**Comment by Colovorat** [comment]
One letter var names 😔

**Comment by TheBoondock** [comment]
> Breaths are done every 8 seconds / 4 life ticks, you must make sure this is either metabolized at the same rate or the numbers are adjusted accordingly to a faster metabolism

Hm this is the same as how healim is done, so wouldnt it be fine as is. Or do you still want me to increase the amount of reagent insert

**Comment by MrMelbert** [comment]
> Hm this is the same as how healium is done...

Metabolism happens once every ~2 seconds (1 life tick) 

Healium heals 2 fire, 2 brute, and 5 toxins every life tick. 
Zauker, in this pr, deals 25 brute, 5 oxy, 8 fire, and 8 toxins every life tick. This is pretty insane

Breaths are only done once every 8 seconds / 4 life ticks, so healium's healing per breath would be 8 fire / 8 brute / 20 toxins
You need to cut zauker's numbers down by 4, otherwise it's too absurd in it's current state

**Comment by timothymtorres** [comment]
> Breaths are only done once every 8 seconds / 4 life ticks, so healium's healing per breath would be 8 fire / 8 brute / 20 toxins You need to cut zauker's numbers down by 4, otherwise it's too absurd in it's current state. Maybe 4 brute, 2 fire, 2 toxins, 1 oxy

Zauker is one of the hardest gases to make.  I main atmos tech and I have only seen it like once or twice in my history of playing.

chrome_NpBi4eYnBM

Because of the requirements it rarely appears in game ever.  Having it be an OP death gas that kills things quickly as a reward is fine.

**Comment by Pickle-Coding** [comment]
Also it decomposes in air very quickly.

**Comment by TheBoondock** [comment]
Ye what tim said, zauker is an incredibly hard to make gas which is why it is insanely deadly and also decomposes in normal air for good measures. Do you still want me to change it @MrMelbert?

**Comment by MrMelbert** [comment]
it doesn't matter if zauker needs to be stronger or not

this PR increases it's damage by 4x you can do that in a separate pr if you feel it needs it

File: code/modules/surgery/organs/lungs.dm [hunk 2]
```diff
@@ -360,10 +360,8 @@
 	// Zauker
 		var/zauker_pp = breath.get_breath_partial_pressure(breath_gases[/datum/gas/zauker][MOLES])
 		if(zauker_pp > gas_stimulation_min)
-			breather.adjustBruteLoss(25)
-			breather.adjustOxyLoss(5)
-			breather.adjustFireLoss(8)
-			breather.adjustToxLoss(8)
+			var/existing= breather.reagents.get_reagent_amount(/datum/reagent/zauker)
```
Comment (MrMelbert) [inline code/modules/surgery/organs/lungs.dm:363]:
```suggestion
			var/existing = breather.reagents.get_reagent_amount(/datum/reagent/zauker)
```

File: code/modules/surgery/organs/lungs.dm [hunk 3]
```diff
@@ -360,10 +360,8 @@
...
+			breather.reagents.add_reagent(/datum/reagent/zauker, max(0,1-existing))
```
Comment (MrMelbert) [inline code/modules/surgery/organs/lungs.dm:364]:
```suggestion
			breather.reagents.add_reagent(/datum/reagent/zauker, max(0, 1 - existing))
```

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 4]
```diff
@@ -1524,6 +1524,22 @@
 	REMOVE_TRAIT(L, TRAIT_RESISTHEAT, type)
 	return ..()
 
+/datum/reagent/zauker
+	name = "Zauker"
+	description = "An unstable gas that is toxic to all living beings"
```
Comment (MrMelbert) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1529]:
```suggestion
	description = "An unstable gas that is toxic to all living beings."
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 5]
```diff
@@ -1524,6 +1524,22 @@
...
+	color = "90560B"
+	taste_description = "bitter"
+	chemical_flags = REAGENT_NO_RANDOM_RECIPE
+
+/datum/reagent/zauker/on_mob_life(mob/living/L, delta_time, times_fired)
+	. = ..()
+	L.adjustBruteLoss(6 * REM * delta_time, FALSE)
+	L.adjustOxyLoss(1 * REM * delta_time, FALSE)
+	L.adjustFireLoss(2 * REM * delta_time, FALSE)
+	L.adjustToxLoss(2 * REM * delta_time, FALSE)
```
Comment (MrMelbert) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1541]: more descriptive variable names - replace `L` with something like `breather` or similar

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 6]
```diff
@@ -1524,6 +1524,22 @@
...
+/datum/reagent/zauker
+	name = "Zauker"
+	description = "An unstable gas that is toxic to all living beings"
+	reagent_state = GAS
+	metabolization_rate = REAGENTS_METABOLISM * 0.5
+	color = "90560B"
+	taste_description = "bitter"
+	chemical_flags = REAGENT_NO_RANDOM_RECIPE
+
+/datum/reagent/zauker/on_mob_life(mob/living/L, delta_time, times_fired)
```
Comment (MrMelbert) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm:1536]: Also, you need to return `TRUE` from this to update their health

File: code/modules/surgery/organs/lungs.dm [hunk 7]
```diff
@@ -360,10 +360,8 @@
 	// Zauker
 		var/zauker_pp = breath.get_breath_partial_pressure(breath_gases[/datum/gas/zauker][MOLES])
 		if(zauker_pp > gas_stimulation_min)
-			breather.adjustBruteLoss(25)
-			breather.adjustOxyLoss(5)
-			breather.adjustFireLoss(8)
-			breather.adjustToxLoss(8)
+			var/existing = breather.reagents.get_reagent_amount(/datum/reagent/zauker)
+			breather.reagents.add_reagent(/datum/reagent/zauker, max(0,1 - existing))
```
Comment (MrMelbert) [inline code/modules/surgery/organs/lungs.dm:364]:
```suggestion
			breather.reagents.add_reagent(/datum/reagent/zauker, max(0, 1 - existing))
```
missed one

**Review by MrMelbert — CHANGES_REQUESTED** [review]
last things I promise

**Comment by TheBonded** [comment]
>Metabolism happens once every ~2 seconds (1 life tick)

@MrMelbert wasnt that halved months ago, to every ~1 second, and chems had their effects multiplied by REM to cut it in half to stay the same as before? 

Or am I entirely wrong and it is in fact 2 seconds(also very likely)?

**Pull request merged**